tests/_helpers/index.ts

```diff
@@ -300,3 +302,29 @@ export function createPrerenderedRoute(
 		[`${file}.rsc.prerender-fallback.rsc`]: `${fileWithBase}.rsc.prerender-fallback.rsc`,
 	};
 }
+
+type ConsoleMethods = 'log' | 'warn' | 'error';
+export function mockConsole<TMethods extends ConsoleMethods>(
+	methods: TMethods[]
```

Comment:
Not sure about the `TMethods[]` that means that the following is valid:
```ts
const mockedConsole = mockConsole(['warn', 'warn', 'warn']);
```
as well as:
```ts
const mockedConsole = mockConsole([]);
```

although they doesn't make sense

we could use `Set<TMethods>` but that feels like it'd makes callers more verbose... (and you could still have an empty set)

You could also define a complex type here to make sure that you always get between one and three methods... but it doesn't feel worth it...

What about `mockConsole(method: ConsoleMethod)` and when you need multiple methods you call the function multiple times?

Reply:
also sorry but I'm really not convinced on the benefits here 😢, using this function requires you to specify the method on creation and then also in the expected calls, feels to me that using it for a single console method is as much work (if not more) as doing things manually (and in almost all cases we do specify a single method)

(the only real benefit I can see is the restoring of all the methods, but again that only makes sense if you have more then one method, which we usually don't)

Reply:
i just wanted a more convenient way to mock the console than doing the full thing each time tbh. i can revert it if you want, or change it to `mockConsole(method: ConsoleMethod)`. i'd probably lean towards the latter rather than just undoing it, because IMO it would still be more convenient, but i dont mind.

Reply:
mh... ok if you feel strongly about the utility I am fine with it 🙂 

as long as after the `mockConsole(method: ConsoleMethod)` you shouldn't have to specify that method again, as in:
```ts
const mockedWarn = mockConsole('warn');
// ...
mockedWarn.expectCalls([
	/.../,
]);
```

would that be ok? 🙂

Reply:
yes that sounds good to me 👍🏻